File: lib-segment/src/test/java/websiteschema/mpsegment/dict/DomainDictionaryTest.java
package websiteschema.mpsegment.dict;

import org.junit.Assert;
import org.junit.Test;
import websiteschema.mpsegment.core.SegmentEngine;
import websiteschema.mpsegment.dict.domain.DomainDictFactory;

import java.util.List;

public class DomainDictionaryTest {

    @Test
    public void should_Loaded_Some_Synonyms_Like_PC() {
        SegmentEngine.getInstance();
        try {
            List<IWord> synonymSet = DomainDictFactory.getInstance().getDomainDictionary().getSynonymSet("个人电脑");
            Assert.assertEquals(synonymSet.get(0).getWordName(), "PC机");
            Assert.assertEquals(synonymSet.get(1).getWordName(), "个人电脑");
        } catch (Exception ex) {
            Assert.fail();
        }
        try {
            List<IWord> synonymSet = DomainDictFactory.getInstance().getDomainDictionary().getSynonymSet("PC机");
            Assert.assertEquals(synonymSet.get(0).getWordName(), "PC机");
            Assert.assertEquals(synonymSet.get(1).getWordName(), "个人电脑");
        } catch (Exception ex) {
            Assert.fail();
        }
    }
}

File: lib-segment-scala/src/test/scala/websiteschema/mpsegment/dict/DomainDictionaryTest.scala
package websiteschema.mpsegment.dict

;

import domain.DomainDictFactory
import org.junit.Assert;
import org.junit.Test;

class DomainDictionaryTest {

  DomainDictFactory().buildDictionary()

  @Test
  def should_Loaded_Some_Synonyms_Like_PC() {
    try {
      var synonymSet = DomainDictFactory().getDomainDictionary().getSynonymSet("个人电脑")
      Assert.assertEquals(synonymSet(0).getWordName(), "PC机");
      Assert.assertEquals(synonymSet(1).getWordName(), "个人电脑");

      synonymSet = DomainDictFactory().getDomainDictionary().getSynonymSet("PC机");
      Assert.assertEquals(synonymSet(0).getWordName(), "PC机");
      Assert.assertEquals(synonymSet(1).getWordName(), "个人电脑");
    } catch {
      case ex: Throwable =>
        Assert.fail();
    }
  }
}
